### PR TITLE
pg_dump support

### PR DESCRIPTION
**Pull request by deem0n**
Allow pg_dump to dump content of the `ag_catalog.*` tables
Fix typo in the `flush_label_oid_cache`

### CODE REVIEW

File: src/backend/utils/cache/ag_cache.c [hunk 1]
```diff
@@ -633,7 +633,7 @@ static void flush_label_oid_cache(void)
 {
     HASH_SEQ_STATUS hash_seq;
 
-    hash_seq_init(&hash_seq, label_name_graph_cache_hash);
+    hash_seq_init(&hash_seq, label_oid_cache_hash);
```
Comment (JoshInnis) [inline src/backend/utils/cache/ag_cache.c:636]: What was your reasoning for doing this? Our goal is to move away from oids?

Comment (deem0n) [reply src/backend/utils/cache/ag_cache.c:636]: We doing tests with dump & restore of the database with AGE extension and some graph data. After restore we have broken oids in the ag_catalog.* tables. Which can be easily fixed and I will commit the fix procedure later. But even after the oid fix we still have errors in the `flush_label_oid_cache(void)`. I think that using `label_name_graph_cache_hash` is just a typo. At least after my fix we see no errors with broken cache anymore after dump & restore and oid fixes

Comment (deem0n) [reply src/backend/utils/cache/ag_cache.c:636]: Eliminating oids would make dump&restore a breeze, but probably will impact query speed ? I did not spent much time in AGE internals to participate in decision making, I guess.

**Comment by emotionbug** [comment]
Looks good to me.

**Comment by JoshInnis** [comment]
@deem0n Please provide a pull request that only includes adding our ag_catalog tables as configuration tables, the saved_result_relinfo issue looks to be a separate issue, given what the next line is. We need to review that as a separate issue from the pg_dump support issue that is the primary change in this PR.

**Comment by jrgemignani** [comment]
Closing this PR as the issues (typos) brought up here have been resolved in commit 0e96fbbf9d6cfb997b8d75da2c8dab1211915319

**Pull request closed without merging**